tests/privacy_scenario_3_test.py

```diff
@@ -0,0 +1,265 @@
+
+    # after last recorded block number (rbn3) is made irreversible, verify 
+    # node2 connects to node1
+    # node2 starts syncing
+    apiNode2.waitForBlock(trans_block_num, reportInterval=1)
+
+    # after node2 is in sync with node1, verify
+    # incoming blocks are sent back to node1 --> for this purpose we use apiNode2 to edit security group
+    trans = securityGroup.editSecurityGroup(removeNodes=[apiNode1], node=apiNode2)
+    # ecord block number (rbn4) transaction was added in
+    trans_block_num = trans["processed"]["block_num"]
+    Print("security group removal of [[\"node3\"]] published on block {}".format(trans_block_num))
+
+    # before last recorded block number (rbn4) is made irreversible, verify
+    # node1 and node2 still stay in sync
+    apiNode1_head = apiNode1.getHeadBlockNum()
+    Print("Node1 head block = {}".format(apiNode1_head))
+    apiNode2_head = apiNode2.getHeadBlockNum()
+    Print("Node2 head block = {}".format(apiNode2_head))
+    apiNode1_InSync = False
+    apiNode2_InSync = False
+    while prod1.getIrreversibleBlockNum() < trans_block_num:
+        temp = apiNode1.getHeadBlockNum()
+        if apiNode1_head < temp:
+            apiNode1_InSync = True
+            apiNode1_head = temp
+        temp = apiNode2.getHeadBlockNum()
+        if apiNode2_head < temp:
+            apiNode2_InSync = True
+            apiNode2_head = temp
+        time.sleep(0.5)
+    
+    Print("Node1 head = {}".format(apiNode1_head))
+    assert trans_block_num < apiNode1_head
+    Print("Node2 head = {}".format(apiNode2_head))
+    assert trans_block_num < apiNode2_head
+    Print("OK: Node1 didn't advance. head still at {}".format(apiNode1_head))
+    assert apiNode1_InSync
+    Print("OK: Node2 didn't advance. head still at {}".format(apiNode2_head))
+    assert apiNode2_InSync
+
+    # after last recorded block number (rbn4) is made irreversible, verify
+    # node1 and node2 do not advance their LIB to rbn4 and do not continue to receive blocks
+    waitSec = 6
+    Print("Waiting for {} seconds to ensure api nodes LIB doesn't move".format(waitSec))
+    time.sleep(waitSec)
+
+    Print("Node1 LIB = {}".format(apiNode1.getIrreversibleBlockNum()))
+    assert apiNode1.getIrreversibleBlockNum() < trans_block_num
+    Print("Node2 LIB = {}".format(apiNode2.getIrreversibleBlockNum()))
+    assert apiNode2.getIrreversibleBlockNum() < trans_block_num
+
+    # add node1 back to security group *
+    trans = securityGroup.editSecurityGroup(addNodes=[apiNode1])
+    # record block number (rbn5) transaction was added in
+    trans_block_num = trans["processed"]["block_num"]
+
+    # before last recorded block number (rbn5) is made irreversible, verify
+    # node1 and node2 do not receive blocks
+    #comparing here with trans_block_num - 12 to avoid possible flakiness when LIB advances inside while
+    while prod1.getIrreversibleBlockNum() <= trans_block_num - 12:
```

Comment:
make similar change to above to do get head on apiNode1 and 2 before getting lib from prod1

Reply:
`apiNode1_head` and `apiNode2_head` was not supposed to move till that moment. Maybe I misunderstood your comment?

Reply:
What I mean is (and similarly for other ones like this) is that if you talk to prod first, and get the number, then talk to the node(s) that are not supposed to change, it can lead to very intermittent failures, since the python script could retrieve the prod1 LIB, then not get processing time for a big gap, in which time prod1 could produce a block, move LIB and send to apiNode1.  I know it seems far fetched, but we have diagnosed similar things in the test framework.  But if you instead do:
while True:
    apiNode1_head = apiNode1.getHeadBlockNum()
    apiNode2_head = apiNode2.getHeadBlockNum()
    prod1Lib = prod1.getIrreversibleBlockNum()
    if prod1Lib >= trans_block_num:
        break
    ...
    
then you are guaranteed that you will stop before apiNode1_head and apiNode2_head should ever advance